dot/state/epoch.go

```diff
@@ -251,14 +253,18 @@ func (s *EpochState) GetEpochData(epoch uint64, header *types.Header) (*types.Ep
 
 	if err != nil && !errors.Is(err, chaindb.ErrKeyNotFound) {
 		return nil, fmt.Errorf("failed to get epoch data from database: %w", err)
-	} else if header == nil {
-		// if no header is given then skip the lookup in-memory
-		return epochData, nil
 	}
 
-	epochData, err = s.getEpochDataFromMemory(epoch, header)
-	if err != nil {
-		return nil, fmt.Errorf("failed to get epoch data from memory: %w", err)
+	//  lookup in-memory only if header is given
+	if header != nil && errors.Is(err, chaindb.ErrKeyNotFound) {
+		epochData, err = s.getEpochDataFromMemory(epoch, header)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get epoch data from memory: %w", err)
+		}
+	}
+
+	if epochData == nil {
+		return nil, errEpochDataNotFound
```

Comment:
nit add context from the args

```suggestion
		return nil, fmt.Errorf("%w: for epoch %d and header with hash %s",
			errEpochDataNotFound, epoch, header.Hash())
```